### DISCUSSION TITLE
Wrong redirects count

### DISCUSSION THREAD

**Discussion by pvtryk** [vercel/next.js#63753] [Help]
### Summary

I'm using redirects in the next.config.js file. I have about 700 records. During the build process, Next.js shows me an error
```
Error: Maximum number of routes (rewrites, redirects, etc) exceeded. Max is 1024, received 1480. Please reduce the number of routes. : https://vercel.com/docs/concepts/limits/overview
```

Despite having about 700 redirects, I encountered this error. While reviewing the Next.js documentation, I couldn't find any information suggesting that redirect counts are doubled. I don't have redirects in another file. If I reduce the number to around 400-450, it works correctly.

single redirect structure:
```
[
    {
      "source": "/page-a/",
      "destination": "/page-b/",
      "permanent": false
    },
    ...
]
```

### Additional information

```js
Next.js: 14.1.4
```

**Comment by icyJoseph** [comment]
Hi,

Looking at:

- https://github.com/vercel/vercel/blob/6487d091df55a1e184de6c63a6a89dbdc18946b8/packages/routing-utils/src/schemas.ts#L56
- https://github.com/vercel/vercel/blob/6487d091df55a1e184de6c63a6a89dbdc18946b8/packages/routing-utils/test/index.spec.ts#L229

It'd seem as if, the base route, and the redirect, do count as two right?

**Comment by youminkim** [comment]
Same issue - We have around 400 routes but it showed the error

```
Error: Maximum number of routes (rewrites, redirects, etc) exceeded. Max is 1024, received 1247. Please reduce the number of routes. : https://vercel.com/docs/concepts/limits/overview
```

In our case, our most of routes are dynamic routes

Next.js version: 14.0.4

**Reply by youminkim** [reply]
Support case created 00221773

Found another case from twitter https://twitter.com/drknature/status/1781188704436113513

cc @ijjk

**Comment by mountainash** [comment]
This was also sending me a little 😵‍💫 for a while... for me (through looking at the Next.js [source code](https://github.com/vercel/next.js/blob/5493d873ab215ae8c82d1334e46164d4d482ee26/packages/next/src/lib/load-custom-routes.ts#L530)) I found it's due to **not** setting `locale: false` when i18n is setup. As per [Redirects with i18n support](https://nextjs.org/docs/pages/api-reference/next-config-js/redirects#redirects-with-i18n-support) it would seem that it's multiplied twice (once for the root and then a regex for combined locale dirs eg. `/en|fr|de/`).

I suggest adding `locale: false` and if you need a match on the locale's use `/:locale/` in the source as the docs suggest.
